### PR TITLE
[WIP]Add MTE-2251 [v124] Add Github Action and Script to Update all Acorn Icons from Source

### PR DESCRIPTION
**Pull request by jjSDET**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/MTE-2251)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/18532)

## **[WIP]**: Pending Requirement :exclamation:
A filter needs to be added to import only the required Assets. The required assets at still unknown, though may initially be what is currently listed in `StandardImageIndentifiers.swift` (to be renamed to `...Identifiers...`).

## :bulb: Description

This pull request introduces an automated workflow to streamline the process of updating icon assets in the Firefox iOS project. Using a custom Python script and GitHub Actions, this workflow significantly reduces manual efforts and ensures consistency in asset updates.

Key Highlights:

- Automated Download & Extraction: The Python script automatically downloads the latest release of icon assets from the FirefoxUX/acorn-icons GitHub repository, extracting relevant files for processing.

- Icon Processing & Placement: The script processes icons in specified directories (16, 20, 24, 30), creating .imageset directories, copying PDF files, and generating Contents.json files in the correct project structure.

- Swift File Update: It updates `StandardImageIndentifiers.swift` with new constants representing these icons, sorted alphabetically and categorized by size (Small, Medium, Large, ExtraLarge).

- GitHub Action Integration: A new GitHub Action has been set up to trigger this script manually. It checks for changes and, if any updates are detected, creates a pull request with the modifications.

- Manual Trigger Feature: This action can be manually triggered via the GitHub Actions tab, offering flexibility for maintenance and updates.

## :pencil: Checklist
You have to check all boxes before merging
- [ ] Filled in the above information (tickets numbers and description of your work)
- [ ] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed I updated documentation / comments for complex code and public methods

### CODE REVIEW

File: .github/workflows/firefox-ios-update-icons.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+name: Update Icons
+
+on: 
+  workflow_dispatch:
```
Comment (isabelrios) [inline .github/workflows/firefox-ios-update-icons.yml:4]: Do we want this to run manually or in a cron?

**Comment by isabelrios** [comment]
This is great, thanks @jjSDET for proposing a solution that fast! 
Let's try to run this locally to double check the solution and then 🚀  (converstaion in slack)

**Comment by thatswinnie** [comment]
Do I understand this correctly that this script will create imagesets for all icons we don't have in the project so far? Or will it just update the existing images? Because we want the latter, as otherwise we would bloat the app size with unused images. 
Also will there be a PR opened for all the changes made? Image changes can have unintended UI side effect and should be tested before the update is merged to main.

**Comment by isabelrios** [comment]
> Do I understand this correctly that this script will create imagesets for all icons we don't have in the project so far? Or will it just update the existing images? Because we want the latter, as otherwise we would bloat the app size with unused images. Also will there be a PR opened for all the changes made? Image changes can have unintended UI side effect and should be tested before the update is merged to main.

I am not sure about the first question, but for the second yes, there is PR created to be reviewed before merging any changes to main.

File: .github/workflows/firefox-ios-update-icons.yml (hunk 1 above)
Comment (clarmso) [reply .github/workflows/firefox-ios-update-icons.yml:4]: We intend to run this Github Actions in a cron. Jackie and I discussed how to test this change on Github Actions before merging this PR. :)

**Comment by jjSDET** [comment]
> Do I understand this correctly that this script will create imagesets for all icons we don't have in the project so far? Or will it just update the existing images? Because we want the latter, as otherwise we would bloat the app size with unused images. Also will there be a PR opened for all the changes made? Image changes can have unintended UI side effect and should be tested before the update is merged to main.

Currently it's setup to create any/all imagesets so we can use a filter to only actually fetch/update the icons we want.

Regarding how we want to filter for what we want, I need to get some more information from you. They could, for example, be passed as build parameters to the Github action, but want to check with you what you'd like to update or add to iOS, if anything.

**Comment by isabelrios** [comment]
I saw the [WIP] added, so I'm converting this to Draft PR for now

**Comment by thatswinnie** [comment]
> > Do I understand this correctly that this script will create imagesets for all icons we don't have in the project so far? Or will it just update the existing images? Because we want the latter, as otherwise we would bloat the app size with unused images. Also will there be a PR opened for all the changes made? Image changes can have unintended UI side effect and should be tested before the update is merged to main.
> 
> Currently it's setup to create any/all imagesets so we can use a filter to only actually fetch/update the icons we want.
> 
> Regarding how we want to filter for what we want, I need to get some more information from you @thatswinnie . They could, for example, be passed as build parameters to the Github action, but want to check with you what you'd like to update or add to iOS, if anything.

All images defined in `StandardImageIdentifiers` should be updated - but only in the sizes they are specified at in that file.

**Pull request closed without merging**